engine/execution/ingestion/engine.go

```diff
@@ -659,6 +709,14 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 	if e.executionDataPruner != nil {
 		e.executionDataPruner.NotifyFulfilledHeight(executableBlock.Height())
 	}
+
+	if e.stopAfterExecuting == executableBlock.ID() {
+		// double check. Even if requested stop height has been changed multiple times,
+		// as long as it matches this block we are safe to terminate
```

Comment:
This is for the case when stop height is changed before the old stop height has been reached.

This leads to think what happens when stop height is changed *after* the old stop height has been reached. 

If EN has stopped at height 100, and then we change the stop height to be 200, would it resume the execution from 100 to 199? It would be complex to get that to work. 

So it seems changing stop height only makes sense if the old stop height has _not_ been reached yet.

If so, how to stop EN from changing stop height if the stop height has been reached?

Reply:
Another great comment, thanks Leo!

You are right that adding any kind of resume would be a lot of work, and can branch into many cases when node it not up to date.

I think best solution would be to disallow change of stop height after any related event happen - so if we skip execution or crash/pause - we should simply lock the change and refuse any change.

Reply:
Yeah, your solution makes sense. 

I wonder what's the use case for pausing the execution when stop height has been reached. If we always just crash, then we eliminate that case, because after stop height has been reached, the node has crashed already, stop height can't be changed again.

Reply:
> I wonder what's the use case for pausing the execution when stop height has been reached. If we always just crash, then we eliminate that case, because after stop height has been reached, the node has crashed already, stop height can't be changed again.

I assume it might be beneficial to keep node running as it serves queries, kind of similar to how we do after spork, when old EN runs this way

Reply:
https://github.com/onflow/flow-go/pull/3213/files#diff-952a99ab2586822131b6723c10c40e350f122df570de319a2b572829ebc34875R62

In StopAtHeight object, that why we have those `Try` functions - they provide synchronisation and set the flag is stopping process has started.